hack/update-codegen.sh

```diff
@@ -7,229 +7,45 @@ set -o pipefail
-    # to be resolved by the end of release-4.15 so this hack can be removed
-    echo "Generating applyconfigurations specifically for MCO"
-    applyconfigurationgen_external_apis_csv="$(codegen::join , "${FQ_APIS[@]}")"
-    applyconfigurations_package="${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-applyconfigurations}"
-    ${APPLYCONFIGURATION_GEN}  \
-      --output-package "${applyconfigurations_package}" \
-      --input-dirs "${applyconfigurationgen_external_apis_csv}" \
-      --external-applyconfigurations k8s.io/api/core/v1.ObjectReference:k8s.io/client-go/applyconfigurations/core/v1 \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.OperatorSpec:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.OperatorStatus:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.OperatorCondition:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.GenerationStatus:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      "$@"
-   else
-    echo "Generating applyconfigurations for $OUTPUT_PKG"
-    applyconfigurationgen_external_apis_csv="$(codegen::join , "${FQ_APIS[@]}")"
-    applyconfigurations_package="${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-applyconfigurations}"
-    ${APPLYCONFIGURATION_GEN}  \
-      --output-package "${applyconfigurations_package}" \
-      --input-dirs "${applyconfigurationgen_external_apis_csv}" \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.OperatorSpec:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.OperatorStatus:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.OperatorCondition:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      --external-applyconfigurations github.com/openshift/api/operator/v1.GenerationStatus:github.com/openshift/client-go/operator/applyconfigurations/operator/v1 \
-      "$@"
-   fi
-
-
-}
-
-# Until we get https://github.com/kubernetes/kubernetes/pull/120877 merged we need to
-# explicitly set these two variables which are not defaulted properly in generate-internal-groups.sh
-export CLIENTSET_PKG=clientset
-export CLIENTSET_NAME=versioned
-
-for group in apiserver apps authorization build cloudnetwork image imageregistry oauth project quota route samples security securityinternal template user; do
-  bash ${CODEGEN_PKG}/generate-groups.sh "lister,informer" \
-    github.com/openshift/client-go/${group} \
-    github.com/openshift/api \
-    "${group}:v1" \
```

Comment:
We no longer need to specify the versions and combinations by the looks of it, has this been tested, everything still generating correctly if we delete an API groups generated clients, which has multiple versions?

Reply:
Based on all my tests, yes we got all the versions properly picked up. I tested that by removing generated files, and making sure new files are generated in replacement.